provider/pkg/schema/gen.go

```diff
@@ -805,12 +820,21 @@ func (ctx *cfSchemaContext) gatherResourceType() error {
 	irreversibleNames := map[string]string{}
 	inputProperties, requiredInputs := map[string]pschema.PropertySpec{}, codegen.NewStringSet()
 	properties, required := map[string]pschema.PropertySpec{}, codegen.NewStringSet()
-	for prop, spec := range ctx.resourceSpec.Properties {
+	props := codegen.SortedKeys(ctx.resourceSpec.Properties)
+	for _, prop := range props {
+		spec := ctx.resourceSpec.Properties[prop]
 		sdkName := naming.ToSdkName(prop)
 		originalSdkName := sdkName
 		if sdkName == "id" {
 			sdkName = "awsId"
 		}
+
+		// augment the documentation with the CloudFormation documentation
+		// we do this before creating the propertySpec because in the process of
+		// creating the propertySpec we change some of the names of properties which
+		// makes it impossible to lookup in the docs
+		ctx.augmentDocumentation(ctx.cfTypeName, prop, spec)
```

Comment:
Are we overloading the `markCreateOnlyProperties` func with too much functionality here? I wouldn't expect that the docs are enhanced within this func

Reply:
I think that's just how the diff folds are displayed. It's part of the `gatherResourceType` function.